src/components/BaseButton.vue

```diff
@@ -1,5 +1,5 @@
 <template>
-    <button :class="buttonClasses" @click="$emit('click')" :type="type">{{caption}}</button>
+    <button :class="buttonClasses" @click.prevent="$emit('click')" :type="type">{{caption}}</button>
```

Comment:
but this change is probably not the best idea - sometimes you may want to have the default behavior.

probably the better idea would be to emit the original event
```
@click.prevent="$emit('click', $event)"
```

then you can prevent it in parent component if you need so for a specific case